### PR TITLE
[KYUUBI #4453][Improvement][K8S] Bump Kubernetes Client Version to 6.4.1

### LINKED ISSUE #4453: [Improvement][K8S] Bump Kubernetes Client Version to 6.4.1
### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### Search before asking

- [X] I have searched in the [issues](https://github.com/apache/kyuubi/issues?q=is%3Aissue) and found no similar issues.

### What would you like to be improved?

Upgrade kubernetes-client from 5.12.1 to 6.4.1.

### How should we improve?

Checking [the compatibility matrix for kubernetes-client](https://github.com/fabric8io/kubernetes-client
), 6.4.1 does not cause incompatibility with lower versions.

And Apache Spark have already bump kubernetes client, here is latest bump PR [[SPARK-42362][BUILD] Upgrade kubernetes-client to 6.4.1](https://github.com/apache/spark/pull/39912)

### Are you willing to submit PR?

- [X] Yes. I would be willing to submit a PR with guidance from the Kyuubi community to improve.
- [ ] No. I cannot submit a PR at this time.

### PR DESCRIPTION
**Pull request by zwangsheng**
### _Why are the changes needed?_

Close #4453

kubernetes client [compare](https://github.com/fabric8io/kubernetes-client/compare/v5.12.1...v6.4.1)

version | K8s 1.26.0 | K8s 1.25.3 | K8s 1.24.7 | K8s 1.23.13 | K8s 1.22.1 | K8s 1.21.1 | K8s 1.20.2 | K8s 1.19.1 | K8s 1.18.0 | K8s 1.17.0 | K8s 1.16.0 | K8s 1.15.3 | K8s 1.14.2 | K8s 1.12.0 | K8s 1.11.0 | K8s 1.10.0 | K8s 1.9.0
--|-- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | --
kubernetes-client 6.4.1 |   | ✓ | ✓ | ✓ | ✓ | ✓ | ✓ | ✓ | ✓ | ✓ | ✓ | ✓ | ✓ | ✓ | ✓ | ✓ | -
kubernetes-client 5.12.1 |   |   |   | ✓ | ✓ | ✓ | ✓ | ✓ | ✓ | ✓ | ✓ | ✓ | ✓ | ✓ | ✓ | ✓ | -

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] [Run test](https://kyuubi.readthedocs.io/en/master/develop_tools/testing.html#running-tests) locally before make a pull request

- [x] [Run ci test]

### CODE REVIEW

**Comment by zwangsheng** [comment]
CI failure not associated with this pr, cc @pan3793 @ulysses-you @turboFei

**Comment by pan3793** [comment]
Please add the supported version range of K8s to the PR description.

**Comment by zwangsheng** [comment]
> Please add the supported version range of K8s to the PR description.

Added

**Review by pan3793 — APPROVED** [review]

File: integration-tests/kyuubi-kubernetes-it/pom.xml [hunk 1]
```diff
@@ -68,6 +68,12 @@
             <scope>test</scope>
         </dependency>
 
+        <dependency>
+            <groupId>io.fabric8</groupId>
+            <artifactId>kubernetes-httpclient-okhttp</artifactId>
+            <scope>test</scope>
+        </dependency>
```
Comment (pan3793) [inline integration-tests/kyuubi-kubernetes-it/pom.xml:75]: why should we need it?

Comment (zwangsheng) [reply integration-tests/kyuubi-kubernetes-it/pom.xml:75]: Fixed `Minikube` use okhttp-client.

Comment (pan3793) [reply integration-tests/kyuubi-kubernetes-it/pom.xml:75]: I don't get your point, it should work w/o introduce extra deps.

Comment (zwangsheng) [reply integration-tests/kyuubi-kubernetes-it/pom.xml:75]: Add comment in parent `pom.xml`,       
> according to kubernetes-client MIGRATION-v6.md, we should include this dependency
   for doing any customization to OkHttp clients.
   https://github.com/fabric8io/kubernetes-client/blob/master/doc/MIGRATION-v6.md#okhttp-httpclient

Comment (pan3793) [reply integration-tests/kyuubi-kubernetes-it/pom.xml:75]: You already defined it in the kyuubi-server module as compile scope, why should define it again?

Comment (zwangsheng) [reply integration-tests/kyuubi-kubernetes-it/pom.xml]: Right, i misunderstand XD, removed.

**Review by turboFei — APPROVED** [review]

**Comment by pan3793** [comment]
Thanks, merged to master

**Comment by pan3793** [comment]
Please open a followup PR to update LICENSE-binary @zwangsheng

**Pull request closed without merging**